### PR TITLE
Appending arbitrary number of Sections/Properties in one go.

### PR DESCRIPTION
**Pull request by msuzen**
This is following review by @jgrewe and @lzehl on the pull request [pull#63](https://github.com/G-Node/python-odml/pull/63)

* `append`  methods on `base` and `sections` modified to handle arbitrary number of arguments.
* Tests are added and they are passing.
* Throwing an error for adding property on a Section, looks a bit more complicated as validation tests are failing in case of this check. See [issue#69](https://github.com/G-Node/python-odml/issues/69)

### CODE REVIEW

File: odml/base.py [hunk 1]
```diff
@@ -136,11 +135,16 @@ def __contains__(self, key):
...
+    def append(self, *obj_tuple):
+        ll = len(obj_tuple)
+        for i in list(range(0,ll)):
+                if obj_tuple[i].name in self:
+                    raise KeyError("Object with the same name already exists! " + str(obj_tuple[i]))
+                # This check gives ERRORS in validator tests
+                # from odml.section import BaseSection
+                # if not isinstance(obj_tuple[i], BaseSection):
+                #	raise KeyError("Object " + str(obj_tuple[i]) + " is not a Section.")
+                super(SmartList, self).append(obj_tuple[i])
```
Comment (jgrewe) [inline odml/base.py]: Why does the commented code raise an error? I would be great to understand the problem

File: odml/base.py [hunk 2]
```diff
@@ -176,10 +180,16 @@ def insert(self, position, section):
...
-        self._sections.append(section)
-        section._parent = self
+        ll = len(vsection)
+        for i in list(range(0,ll)):
+	     # from odml.section import BaseSection
+             # This check gives ERRORS in validator tests
+             #if isinstance(vsection[i], BaseSection):
+             #	raise KeyError("Object " + str(vsection[i]) + " is not a Section.")
+             self._sections.append(vsection[i])
+             vsection[i]._parent = self
```
Comment (jgrewe) [inline odml/base.py]: same as above

File: odml/base.py [hunk 3]
```diff
@@ -176,10 +180,16 @@ def insert(self, position, section):
         self._sections.append(section)
         section._parent = self
 
-    def append(self, section):
+    def append(self, *vsection):
         """adds the section to the section-list and makes this document the section’s parent"""
-        self._sections.append(section)
```
Comment (jgrewe) [inline odml/base.py]: I guess, the doctoring should be updated

File: odml/base.py (hunk 2 above)
Comment (msuzen) [inline odml/base.py]: @jgrewe Thank you for the detailed review. Yes, I should decrease the number of commits :) It is an old habit and due to I build the branch over git at each new test. 

Regarding validator test errors in detecting attempted addition of a property to the main Document object, this is also the issue #69. I will get back on this.

Comment (jgrewe) [inline odml/base.py]: cool! thanks

**Comment by msuzen** [comment]
@jgrewe Regarding `import` issue within a method, it stems from having #69 as a requirement creates circular class dependency, i.e., `section.py` requiring `odml.base` and  `base.py` requiring `odml.section`, similarly for `doc.py`. 

Might be a [Code Smell](http://stackoverflow.com/questions/1356304/are-circular-class-dependencies-bad-from-a-coding-style-point-of-view)  : ) I think you were mentioning recently on refactoring classes.

**Pull request closed without merging**